ibis/backends/impala/__init__.py

```diff
@@ -259,16 +256,21 @@ def raw_sql(self, query: str):
 
         return cursor
 
-    def fetch_from_cursor(self, cursor, schema):
+    def _fetch_from_cursor(self, cursor, schema):
+        from ibis.formats.pandas import PandasData
+
         results = fetchall(cursor)
-        if schema:
-            return PandasData.convert_table(results, schema)
-        return results
+        return PandasData.convert_table(results, schema)
 
     @contextlib.contextmanager
     def _safe_raw_sql(self, query: str | DDL | DML):
         if not isinstance(query, str):
-            query = query.compile()
+            try:
+                query = query.sql(dialect=self.compiler.dialect)
+            except AttributeError:
+                query = query.compile()
```

Comment:
Is this for the `base/sql/ddl.py` objects?

Reply:
Yes.